packages/permit2-sdk/src/constants.ts

```diff
@@ -20,7 +20,7 @@ const PERMIT2_ADDRESSES: { [key: number]: Address } = {
   [ChainId.ZKSYNC_TESTNET]: '0x0000000000000000000000000000000000000000',
 }
 
-export const PERMIT2_ADDRESS = (chainId: number | undefined): Address => {
+export const getPermit2Address = (chainId: ChainId | undefined): Address => {
   if (chainId === undefined) return PERMIT2_ADDRESSES[56]
```

Comment:
```suggestion
  if (typeof chainId === 'undefined') return PERMIT2_ADDRESSES[56]
```

Reply:
for es5 and local variable I think it's ok to simply compare with undefined?

Reply:
another thing is
```suggestion
  if (chainId === undefined) return PERMIT2_ADDRESSES[ChainId.BSC]
```

Reply:
I think always use `typeof` is better, unify the code style

if value is undeclared variable, will throw a error

Reply:
yea but when we're talking about undeclared variable, we normally refer to global variable. It's not applicable in this case

Reply:
for your comment above jackson on defaulting to BSC is chainID IS undefined. i do this to fix a ts error check that appears from use of getPermitAddress and getUniversalRouter in the SwapCommitButton. my justification was that if chainId is undefined, it means that user is not connected or else for a split second between switching chains, so its fine to use this default fallback since user will never be able to swap if chainId is undefined, so this default case will never get triggered when an actual swap occurs

Reply:
it's ok to default to bsc. My point is let's use `ChainId` instead of number literal

Reply:
ahhh i see i get you. yep changing type

Reply:
Late in the party 😅
I think we should better throw error on library/sdk side and let applications aware and handle this case

Reply:
so will i keep the error then